StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1518CodeMustNotContainBlankLinesAtEndOfFile.cs

```diff
@@ -43,7 +44,17 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
         /// <inheritdoc/>
         public override void Initialize(AnalysisContext context)
         {
-            // TODO: Implement analysis
+            context.RegisterSyntaxTreeAction(this.HandleSyntaxTreeAction);
+        }
+
+        private void HandleSyntaxTreeAction(SyntaxTreeAnalysisContext context)
+        {
+            var lastToken = context.Tree.GetRoot().GetLastToken(includeZeroWidth: true);
+
+            if (lastToken.HasLeadingTrivia)
```

Comment:
Not all trivias are whitespaces. A comment at the end of the file is reported too or the endif here:
```csharp
#if true
namespace Foo
{
}
#endif
```